app/models/location.rb

```diff
@@ -134,6 +134,19 @@ class Location < AbstractModel
     end
   end
 
+  include CreatedUpdatedScopes
+
+  scope :name_like,
+        ->(name) { where(Location[:name].matches("%#{name}%")) }
```

Comment:
Would it be better to leave off the "%" and have the caller add those?  I see two arguments for, one against -- it makes it more clear whether the "%" are there or not (they will be explicitly added to every call, as opposed to invisibly and implicitly added where you will only know they're added if you check the implementation); and it allows for cases where you do not want one or both of the percents.  But if every single call in practice adds them, it of course will involve a great deal more work to add them explicitly every time.  If this latter case is the most important factor, how about changing it from "name_like" to "name_includes" or "name_with/has_substring" or something like that instead?  Yes, it hides the fact that it's using "LIKE" and all the features that opens up, but if all the actual use cases don't actually *need* any of the features of LIKE anyway...

Reply:
Good question. Goooood question. I thought about it, but attempted it too late at night and couldn't keep the `% %` straight and reverted - there are so many uses in the tests of these. But I like the idea of the caller adding them specifically. I can attempt again when fresher.

I definitely prefer `name_includes` to `name_like`.  `name_matches` would also be better than `name_like`, but maybe more algorithmic-sounding? It does suggest the fact that "agaricus" will also match "nimmoagaricus". Hmm...

Reply:
I feel ok with having the caller just enter the text and having the scope add the `%`s appropriate to the scope, at least for now. 

Joe pointed out that the trailing `%` should be eliminated from the `scope :in_region` though, because the region is currently expected to be anchored to the end of the string. (I guess `%string%` implies there is _definitely_ another character after the matched pattern?) I committed this change already.

Maybe we should consider the versatility of that — what if the region entered is "California" or "Normandie", i.e. definitely a unique region but no country given? I guess we do force people to add the country to get a match.

Reply:
1. % can match the empty string, too.  "%string%" does not imply there is another character after (or before) "string".

2. re: region being anchored to the end -- Yeah, it would be nice to be able to just say "California", but what about Baja California and maybe other Californias elsewhere in the Americas?  To be precise it really needs to be anchored at the end.  If you want the UI to be more convenient, I think the "middleman" needs to add in the implicit ", USA" and pass that into the scope.  I don't think we can expect the scope to be that smart.

(Good catch, by the way, Joe!  I totally missed the region anchor, and the sequence scopes in my read-through.)